internal/controller/controller.go

```diff
@@ -288,6 +293,69 @@ func (c *httpController) exportRecordedData(writer http.ResponseWriter, request
 // importRecordedData imports data from a previously exported record session.
 // An error is returned if a record or replay session is currently running or the data is incomplete
 func (c *httpController) importRecordedData(writer http.ResponseWriter, request *http.Request) {
-	//TODO implement me using TDD
-	writer.WriteHeader(http.StatusNotImplemented)
+	importedRecordedData := &dtos.RecordedData{}
+	var reader io.ReadCloser
+	var err error
+	var overWriteProfilesDevices bool
+
+	compression := request.URL.Query().Get("compression")
+	overWriteProfilesDevices, err = strconv.ParseBool(request.URL.Query().Get("overwrite"))
+	if err != nil {
+		writer.WriteHeader(http.StatusInternalServerError)
+		_, _ = writer.Write([]byte(fmt.Sprintf("failed to parse overwrite parameter: %v", err)))
+	}
+
+	if len(compression) == 0 {
+		err = json.NewDecoder(request.Body).Decode(&importedRecordedData)
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			_, _ = writer.Write([]byte(fmt.Sprintf("%s: %v", failedRequestJSON, err)))
+			return
+		}
+	} else {
+		switch compression {
+		case "GZIP":
+			reader, err = gzip.NewReader(request.Body)
+			if err != nil {
+				writer.WriteHeader(http.StatusInternalServerError)
+				_, _ = writer.Write([]byte(fmt.Sprintf("%s: %s", failedToUncompressData, err)))
+				return
+			}
+
+		case "ZLIB":
+			reader, err = zlib.NewReader(request.Body)
+			if err != nil {
+				writer.WriteHeader(http.StatusInternalServerError)
+				_, _ = writer.Write([]byte(fmt.Sprintf("%s: %s", failedToUncompressData, err)))
+				return
+			}
+		default:
+			writer.WriteHeader(http.StatusInternalServerError)
+			_, _ = writer.Write([]byte("compression format not supported"))
+			return
+
+		}
+		defer reader.Close()
+		err = json.NewDecoder(reader).Decode(&importedRecordedData)
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			_, _ = writer.Write([]byte(fmt.Sprintf("%s: %v", failedRequestJSON, err)))
+			return
+		}
+	}
+
+	emptydata := &dtos.RecordedData{}
+	if reflect.DeepEqual(importedRecordedData, emptydata) {
+		writer.WriteHeader(http.StatusNoContent)
+		_, _ = writer.Write([]byte(noDataFound))
+		return
+	}
```

Comment:
I think we should be more detailed here. I.e. first check for Events not Empty  (need at least 1) , then Devices not empty (need at least 1) and then Profiles not Empty (need at least 1) . This give better feedback to user. Also should return `BadRequest` in this situation